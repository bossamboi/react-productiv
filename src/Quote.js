import React from "react";

function Quote({ quote = null }) {
	const quoteHTML = quote && (
		<p className="Quote-quote">
			<i>
				{quote.text} -{quote.author}
			</i>
		</p>
	);

	return <div>{quoteHTML}</div>;
}

export default Quote;
