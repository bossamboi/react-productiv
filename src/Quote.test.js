import React from "react";
import { render } from "@testing-library/react";
import Quote from "./Quote";

describe("Quote container", function () {
	const testQuote = {
		text: "Test text",
		author: "Testy Testman",
	};

	it("renders without crashing", function () {
		render(<Quote quote={testQuote} />);
	});

	it("renders expected text", function () {
		const result = render(<Quote quote={testQuote} />);
		expect(result.queryByText("Test text -Testy Testman")).toBeInTheDocument();
	});

	it("does not render anything when no quote is passed", function () {
		const { container } = render(<Quote />);
		expect(container.querySelector(".Quote-quote")).not.toBeInTheDocument();
	});

	it("matches snapshot", function () {
		const { container } = render(<Quote quote={testQuote} />);
		expect(container).toMatchSnapshot();
	});
});
