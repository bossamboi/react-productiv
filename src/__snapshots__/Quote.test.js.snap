// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote container matches snapshot 1`] = `
<div>
  <div>
    <p
      class="Quote-quote"
    >
      <i>
        Test text
         -
        Testy Testman
      </i>
    </p>
  </div>
</div>
`;
